packages/p2p-v2/src/components/Modals/NicknameModal/NicknameModal.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { ComponentType, useEffect } from 'react';
+import { Controller, useForm, useWatch } from 'react-hook-form';
+import Modal from 'react-modal';
+import { p2p } from '@deriv/api';
+import { Button } from '@deriv-com/ui/dist/components/Button';
+import { Text } from '@deriv-com/ui/dist/components/Text';
+import { useDevice } from '../../../hooks';
+import P2PUserIcon from '../../../public/ic-cashier-p2p-user.svg';
+import { Input } from '../../Input';
+import { customStyles } from '../helpers';
+import './NicknameModal.scss';
+
+const NicknameModal = () => {
+    const ReactModal = Modal as ComponentType<ReactModal['props']>;
+    const {
+        control,
+        formState: { errors },
+        getValues,
+        handleSubmit,
+    } = useForm({
+        defaultValues: {
+            nickname: '',
+        },
+        mode: 'onChange',
+    });
+
+    const { error, isError, mutate, reset } = p2p.advertiser.useCreate();
+    const { isMobile } = useDevice();
+    const textSize = isMobile ? 'md' : 'sm';
+
+    const onSubmit = () => {
+        mutate({ name: getValues('nickname') });
+        setTimeout(() => {
+            reset();
+        }, 3000);
```

Comment:
@ameerul-deriv, isn't it possible to call `reset` in something like a `useEffect` that tracks the `success` status of the `mutation`, as opposed to using `setTimeout`?

Reply:
hmmm, because the response from the mutation wont change when theres an error, it will be stuck for a long time, and the error message will not disappear. Hence why I added a timeout to have the message there temporarily. 

The behaviour of the button is that if theres any error, it'll be disabled, and mutation error won't reset unless we explicitly call it

Reply:
Oooh okay, but why not use `debounce` at least?

Reply:
ah yeah true, will do that